pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -227,8 +227,16 @@ func (a *AgentClusterInstall) Generate(dependencies asset.Parents) error {
 					VSphere: &vspherePlatform,
 				}
 			}
-			agentClusterInstall.Spec.APIVIP = installConfig.Config.Platform.VSphere.APIVIPs[0]
-			agentClusterInstall.Spec.IngressVIP = installConfig.Config.Platform.VSphere.IngressVIPs[0]
+			if len(installConfig.Config.Platform.VSphere.APIVIPs) == 0 || len(installConfig.Config.Platform.VSphere.IngressVIPs) == 0 {
```

Comment:
To rule out the case that a user fails to enter either field and didn't intend to use UserManagedNetworking.
```suggestion
			if len(installConfig.Config.Platform.VSphere.APIVIPs) == 0 && len(installConfig.Config.Platform.VSphere.IngressVIPs) == 0 {
```